### PR TITLE
test(server-test): mock errorHandler using jest.fn()

### PR DESCRIPTION
**Pull request by oscard0m**
- [x] Use `jest.fn()` for `errorHandler` mock
- [x] Pending to solve `req.body` TypeScript error
- [x] Timeouts

### CODE REVIEW

File: test/integration/server-test.ts [hunk 1]
```diff
@@ -98,6 +97,11 @@ test("POST / with push event payload (request.body already parsed)", (t) => {
   const server = http.createServer((req, res) => {
     req.once("data", (chunk) => dataChunks.push(chunk));
     req.once("end", () => {
+      /*
+       * @ts-ignore: Until getPayload @ts-ignore is fixed
+       * (https://github.com/dominguezcelada/webhooks.js/blob/b2162f4489efb9ee828cfd57e82bf98d12503c4b/src/middleware/get-payload.ts#L7-L8)
+       */
+      //@ts-ignore
       req.body = JSON.parse(Buffer.concat(dataChunks).toString());
```
Comment (oscard0m) [inline test/integration/server-test.ts]: ts-ignored until this is fixed:

https://github.com/dominguezcelada/webhooks.js/blob/b2162f4489efb9ee828cfd57e82bf98d12503c4b/src/middleware/get-payload.ts#L7-L8

File: package.json [hunk 2]
```diff
@@ -69,6 +69,7 @@
     "preset": "ts-jest",
     "testMatch": null,
     "testRegex": "test/.*/.*.ts",
+    "testEnvironment": "node",
```
Comment (oscard0m) [inline package.json]: Avoid timeouts when running `axios` with `jest` Context: https://github.com/axios/axios/issues/1754#issuecomment-435784235

File: test/integration/server-test.ts [hunk 3]
```diff
@@ -97,6 +97,11 @@ test("POST / with push event payload (request.body already parsed)", (t) => {
   const server = http.createServer((req, res) => {
     req.once("data", (chunk) => dataChunks.push(chunk));
     req.once("end", () => {
+      /*
+       * @ts-ignore: Until getPayload @ts-ignore is fixed
+       * (https://github.com/dominguezcelada/webhooks.js/blob/b2162f4489efb9ee828cfd57e82bf98d12503c4b/src/middleware/get-payload.ts#L7-L8)
+       */
+      //@ts-ignore
```
Comment (gr2m) [inline test/integration/server-test.ts]: it's not really something that needs fixing, it's a use case with have to account for. `req.body` is set in some serverless environments, the parsing is happening outside of our control. 

We could amend the `@ts-ignore in both places
```suggestion
      // @ts-ignore account for serverless environments that provide a pre-parsed `req.body` object
```

**Comment by wolfy1339** [comment]
I think you have a bad merge/rebase going on here.

8cbe6a5  shouldn't be there

**Comment by oscard0m** [comment]
> I think you have a bad merge/rebase going on here.
> 
> [8cbe6a5](https://github.com/octokit/webhooks.js/commit/8cbe6a5d2ec0c2e41af3fb49f5717b556980e9a4) shouldn't be there

Tried to rebase again. Checking the files changed seems to be only a new line and that's it right? I will be afk next days, just FYI

**Comment by wolfy1339** [comment]
@dominguezcelada I fixed it. Please double check all the changes you intended are still here.

Next time I won't rebase the base branch when other people are submitting PRs to it :laughing: I'll do it when it's ready to be merged

**Comment by oscard0m** [comment]
> @dominguezcelada I fixed it. Please double check all the changes you intended are still here.
> 

On the changes I remember I think I left a comment regarding @ts-ignore for req.body line and also some `test(done)` to control some async tests. I think today night (Barcelona time) I will have access to my laptop and double check my local changes.

> 
> Next time I won't rebase the base branch when other people are submitting PRs to it :laughing: I'll do it when it's ready to be merged

No problem. And thanks!

**Comment by wolfy1339** [comment]
Hey @dominguezcelada did you manage to double check this?

**Pull request merged**